3 - PB/Documentazione esterna/Specifica Tecnica/Specifica Tecnica.typ

```diff
@@ -255,7 +256,391 @@ Ciò avviene prestando attenzione alle tecnologie impiegate sia nel front-end ch
 
 == Descrizione generale
 
-== Diagramma delle classi
+== Struttura
+Il software WMS3 al fine di perseguire manutenibilità, flessibilità e scalabilità, adotta ed implementa un'architettura "layered", nota anche come "Multi-tier architecture".
+Tale architettura permette di individuare e suddividere la logica del software in 3 principali aspetti, definiti tier (separation of concerns), quali:
+
+- *Persistence layer*: gestisce l'accesso al database e fornisce gli strumenti dedicati alla lettura dei dati al suo interno. I dati letti vengono processati al fine di poter creare gli elementi del Business layer;
+
+- *Business layer*: si occupa di elaborare i dati ricevuti dal layer di persistenza e applicare le regole di business definite. È responsabile di implementare la logica dell'applicazione in modo indipendente dalle tecnologie di persistenza e di presentazione utilizzate;
+
+- *Presentation layer*: permette di trasformare i dati elaborati dal Business layer e le informazioni in una forma comprensibile e accessibile agli utenti finali. Questo include la creazione di interfacce utente grafiche e visualizzazioni 3D degli elementi di interesse.
+
```

Comment:
Aggiungere specifica che i nostri layer sono closed. Pertanto il passaggio delle informazioni tra due layer diversi passa attraverso tutti i layer in mezzo (best practice), quindi non esiste che la Presentation comunichi direttamente con il persistance o viceversa.

Maggiori dettagli: https://www.math.unipd.it/~rcardin/swea/2022/Software%20Architecture%20Patterns.pdf

Non serve dire molto, ma è bene menzionarlo dai